aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -557,8 +568,12 @@ static inline void shapeCheck3d(
   int dimd = 1;
   int dimslices = 0;
 
-  TORCH_CHECK(input.numel() > 0 && (input.dim() == 4 || input.dim() == 5),
-      "non-empty 4D or 5D (batch mode) tensor expected for input, but got: ", input);
+  // allow batch size of 0-dim.
+  bool valid_dims = input.size(1) != 0 && input.size(2) != 0 && input.size(3) != 0;
```

Comment:
for 4d tensor dimensions 0,1,2,3 should be nonzero (there's no batch dimension, so tensor should be non-empty), for 5d tensor dimensions 1,2,3,4 should be nonzero.